analyzer/codechecker_analyzer/analyzers/gcc/analyzer.py

```diff
@@ -0,0 +1,254 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+"""
+"""
+from collections import defaultdict
+# TODO distutils will be removed in python3.12
+from distutils.version import StrictVersion
+import os
+import pickle
+import shlex
+import subprocess
+
+from codechecker_common.logger import get_logger
+
+from codechecker_analyzer import analyzer_context
+
+from .. import analyzer_base
+from ..flag import has_flag
+from ..config_handler import CheckerState
+
+from .config_handler import GccConfigHandler
+from .result_handler import GccResultHandler, \
+    actual_name_to_codechecker_name, codechecker_name_to_actual_name_disabled
+
+LOG = get_logger('analyzer.gcc')
+
+
+class Gcc(analyzer_base.SourceAnalyzer):
+    """
+    Constructs the Gcc analyzer commands.
+    """
+
+    ANALYZER_NAME = 'gcc'
+
+    @classmethod
+    def analyzer_binary(cls):
+        return analyzer_context.get_context() \
+            .analyzer_binaries[cls.ANALYZER_NAME]
+
+    @classmethod
+    def get_version(cls, env=None):
+        """ Get analyzer version information. """
+        return cls.__get_analyzer_version(cls.analyzer_binary(), env)
+
+    def add_checker_config(self, checker_cfg):
+        LOG.warning("Checker configuration for Gcc is not implemented yet")
```

Comment:
I think even a warning is not needed here. Simply don't provide any gcc analyzer checker options.

In the latest implementation I tested I even got an error here

[ERROR 2023-10-12 19:22] - Failed to get checker configuration options for 'gcc' analyzer(s)! Please try to upgrade your analyzer version to use this feature.

but actually this is not cause for a warning or an error.
Simply we dont provide any checker option yet.